### PR TITLE
Merge in defaults from subcommand for the parser description.

### PR DESCRIPTION
**Pull request by dereuromark**
### Before
```php
        $parser->addSubcommand('infos', [
            'help' => 'Get some GitHub infos',
            'parser' => [
                //'description' => 'Get some GitHub infos' // Without declaring this again
            ],
        ]);
```

results in no --help output (displaying main page again) and
```
Usage:
cake release [subcommand] [-d] [-h] [-q] [-v]

Subcommands:

infos  Get some GitHub infos
pr     Release PR

To see help on a subcommand use `cake release [subcommand] --help`
```

It only works if you manually copy and paste the help text into the parser description (unDRY).

### After
The above results in expected

```
Get some GitHub infos

Usage:
cake release infos [-h] [-q] [-v]

Options:

--help, -h     Display this help.
--quiet, -q    Enable quiet output.
--verbose, -v  Enable verbose output.
```

### CODE REVIEW

File: src/Console/ConsoleOptionParser.php [hunk 1]
```diff
@@ -589,6 +592,39 @@ public function addSubcommand($name, array $options = [])
...
+     * @param array $options Options
+     * @return array
+     */
+    protected function _mergeSubcommandHelpToParserDescription($options)
+    {
+        if (!$options['help']) {
+            return $options;
+        }
+
+        if ($options['parser'] && is_object($options['parser'])) {
```
Comment (markstory) [inline src/Console/ConsoleOptionParser.php]: Shouldn't you be checking the interface/that the object has the method you want to call?

Comment (dereuromark) [inline src/Console/ConsoleOptionParser.php]: We could check on the interface as well, I just assumed that if it is an object it must be the expected one.

Comment (dereuromark) [inline src/Console/ConsoleOptionParser.php]: ConsoleOptionParser doesnt have an interface, other code checks for the concrete class.
Maybe we could add an interface for a future release here? That would be great.

Comment (markstory) [inline src/Console/ConsoleOptionParser.php]: or use `method_exists()` 😄 An interface would work too though.

Comment (dereuromark) [inline src/Console/ConsoleOptionParser.php]: I think the PR is fine now as it is (as other code pieces use the same check).

Comment (markstory) [inline src/Console/ConsoleOptionParser.php]: ok.

**Pull request merged**